Source/SourceKittenFramework/DyldHelper.swift

```diff
@@ -0,0 +1,192 @@
+
+/// Environment Variable that indicates current process is launched through `execv()`
+private let sourceKittenFrameworkLoadingkey = "SOURCEKITTEN_FRAMEWORK_LOADING"
+
+private let libclangPath = "libclang.dylib"
+private let sourcekitdPath = "sourcekitd.framework/Versions/A/sourcekitd"
+
+/// Checks whether depending dynamic link libraries are loaded or not.
+/// If not, coordinates `DYLD_*` environment variables, calls `execv()` and never returns.
+public func checkDepndenciesOfSourceKittenFramework() {
+    if libraryIsLoaded(libclangPath) && libraryIsLoaded(sourcekitdPath) {
+        return
+    }
+
+    // following codes will be executed when dependencies will not be loaded.
+    // On sourcekitten installation, LC_RPATH of SourceKittenFramework does not contain pathes
+    // which can find libclang.dylib and sourcekitd.framework.
+
+    coordinateEnvironmentVariablesForDYLD()
+    setenv(sourceKittenFrameworkLoadingkey, "1", 1)
+
+    let arguments = NSProcessInfo.processInfo().arguments
+    let path = arguments[0]
+    let argv = CStringArray(arguments)
+    execv(path, argv.pointers) // never returns on normal flow
+    let error = String(UTF8String: strerror(errno))! // swiftlint:disable:this force_unwrapping
+    fatalError("execv failed: \(error)")
+}
+
+/// Returns true if library is not loaded.
+/// - Parameter path: string of library's path
+private func libraryIsLoaded(path: String) -> Bool {
+    let library = dlopen(path, RTLD_LAZY)
+    if library != nil {
+        dlclose(library)
+        return true
+    } else  if let _ = NSProcessInfo.processInfo().environment
+        .indexForKey(sourceKittenFrameworkLoadingkey) {
+        fatalError("Can't load \(path)")
+    }
+    return false
+}
+
+/// Coordinates "DYLD_LIBRARY_PATH" and "DYLD_FRAMEWORK_PATH" for `execv`ed process.
+private func coordinateEnvironmentVariablesForDYLD() {
+    let fileManager = NSFileManager.defaultManager()
+    let developerDirXcode = "/Applications/Xcode.app/Contents/Developer"
+    let developerDirXcodeBeta = "/Applications/Xcode-beta.app/Contents/Developer"
+
+    var pathes: [String] = [
+        toolchainDir,
+        xcodeSelectPath.map(toolchainDirFrom),
+        /*
+        Followings are used when `xcode-select -p` points "Command Line Tools OS X for Xcode",
+        but Xcode.app exists.
+        */
+        toolchainDirFrom(developerDirXcode),
+        toolchainDirFrom(developerDirXcodeBeta),
+        toolchainDirFrom(homeDir.stringByAppendingPathComponent(developerDirXcode)),
+        toolchainDirFrom(homeDir.stringByAppendingPathComponent(developerDirXcodeBeta)),
+        ].flatMap { path in
+            if let fullPath = path?.stringByAppendingPathComponent("/usr/lib")
+                where fileManager.fileExistsAtPath(fullPath) {
+                    return fullPath
+            }
+            return nil
+        }
+
+    #if SWIFT_PACKAGE
+        pathes = ["/Library/Developer/Toolchains/swift-latest"] + pathes
+    #endif
```

Comment:
`/Applications/` and `/Library/Developer/` are not guaranteed constants by the platform; Foundation's search path APIs should be used instead. Additionally, Swift toolchains can be installed to `~/Library/Developer`.